homework/index.js

```diff
@@ -30,13 +30,149 @@
+    document.getElementById('section1').innerHTML = htmlString;
+  }
+
+  // Create options under 'SELECT' element which should have HYF Repositories.
+
+  function loadSelectionValues(userInfo) {
+    const sortRepoName = [];
+
+    const userRepo = JSON.parse(userInfo);
+    const mySelect = document.getElementById('myselect');
+
+    // push all the HYP repo names to sort array.
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const repo in userRepo) {
+      if (Object.prototype.hasOwnProperty.call(userRepo, repo)) {
+        sortRepoName.push(userRepo[repo].name);
+      }
+    }
+
+    // sort the repo name using sort function and localeComapare for uppercase and lowercase sorting.
+    sortRepoName.sort((a, b) => a.localeCompare(b));
+
+    // Create Option under Select element and attach the  same with SELECT element.
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const repo of sortRepoName) {
+      const option = document.createElement('option');
+      option.value = repo;
+      option.text = repo;
+      mySelect.appendChild(option);
+    }
+
+    const selectBox = document.getElementById('myselect');
+    const selectedValue = selectBox.options[selectBox.selectedIndex].value;
+
+    // Load Repository information for the choose repository name in the select box.
+    loadDiv(userRepo, selectedValue);
+    selectBox.onchange = function() {
+      loadDiv(userRepo, selectBox.value);
+    };
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
+      const myParent = document.body;
+
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        // Create div element 'select' in  document body to hold the label element and list box.
+
+        createAndAppend('div', myParent, { id: 'select' });
+        const select = document.getElementById('select');
+        createAndAppend('LABEL', select, {
+          text: 'HYF Repositories: ',
+          id: 'label',
+          for: 'repo',
+        });
+        createAndAppend('select', select, { id: 'myselect' });
+
+        // Create two div elements section1 and section2 under 'Root' div to have
+        // section1 - Repository Information.
+        // section2 - Contributions.
+
+        createAndAppend('div', root, { id: 'section1' });
+        createAndAppend('div', root, { id: 'section2' });
+
+        // Insert section1 before section2 div element under 'root' div.
+        const newNode = document.getElementById('section2');
+        const referenceNode = document.querySelector('section1');
+        root.insertBefore(newNode, referenceNode);
+
+        // Insert Select div first in the body before root div.
+        myParent.insertBefore(select, document.getElementById('root'));
+
+        loadSelectionValues(JSON.stringify(data, null, 2));
```

Comment:
Does it need to be stringified? Kinda confused what the other two parameters do.

Reply:
It was there in existing code, but i corrected ans used the data as such.